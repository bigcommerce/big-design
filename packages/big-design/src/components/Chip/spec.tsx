import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import React from 'react';
import 'jest-styled-components';

import { Chip } from './index';

test('renders the label', () => {
  const label = 'Test';

  render(<Chip label={label} />);

  expect(screen.getByText(label)).toBeInTheDocument();
});

test('renders without close button', () => {
  render(<Chip data-testid="chip" label="Test" />);

  expect(screen.queryByRole('button')).not.toBeInTheDocument();
  expect(screen.getByTestId('chip')).toMatchSnapshot();
});

test('renders with close button if onRemove is present', () => {
  render(<Chip data-testid="chip" label="Test" onDelete={jest.fn()} />);

  expect(screen.getByRole('button')).toBeInTheDocument();
  expect(screen.getByTestId('chip')).toMatchSnapshot();
});

test('onDelete is called when close button is clicked', async () => {
  const onDelete = jest.fn();

  render(<Chip label="Test" onDelete={onDelete} />);

  await userEvent.click(screen.getByRole('button'));

  expect(onDelete).toHaveBeenCalled();
});

test('accepts custom margin props', () => {
  render(<Chip data-testid="chip" label="Test" margin="large" onDelete={jest.fn()} />);

  expect(screen.getByTestId('chip')).toHaveStyle('margin: 1.25rem');
});
