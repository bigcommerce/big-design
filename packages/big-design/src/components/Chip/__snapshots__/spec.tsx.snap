// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders with close button if onRemove is present 1`] = `
.c6 {
  vertical-align: middle;
  height: 1rem;
  width: 1rem;
}

.c2 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
  margin: 0;
  margin-right: 0.25rem;
}

.c2:last-child {
  margin-bottom: 0;
}

.c0 {
  margin: 0.25rem;
  padding-right: 0.25rem;
  padding-left: 0.5rem;
  box-sizing: border-box;
  background-color: #D9DCE9;
  border-radius: 0.25rem;
}

.c3 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: background-color,border-color,box-shadow,color;
  transition-property: background-color,border-color,box-shadow,color;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  background-color: transparent;
  border-color: transparent;
  color: #3C64F4;
}

.c3:focus {
  outline: none;
}

.c3[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c3 + .bd-button {
  margin-top: 0.5rem;
}

.c3:active {
  background-color: #DBE3FE;
}

.c3:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c3:hover:not(:active) {
  background-color: #F0F3FF;
}

.c3[disabled] {
  border-color: transparent;
  color: #D9DCE9;
}

.c5 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4 {
  color: #5E637A;
  height: auto;
  padding: 0;
  width: auto;
  min-width: auto;
}

@media (min-width:720px) {
  .c3 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c3 {
    width: auto;
    padding: 0;
    min-width: 2.25rem;
  }
}

<div
  class="c0 c1"
  data-testid="chip"
>
  <p
    class="c2"
  >
    Test
  </p>
  <button
    aria-label="Remove Test"
    class="c3 c4"
    type="button"
  >
    <span
      class="c5"
    >
      <svg
        aria-labelledby=":r0:"
        class="c6"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <title
          id=":r0:"
        >
          Delete
        </title>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`renders without close button 1`] = `
.c2 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
  margin: 0;
  margin-right: 0.25rem;
}

.c2:last-child {
  margin-bottom: 0;
}

.c0 {
  margin: 0.25rem;
  padding-right: 0.25rem;
  padding-left: 0.5rem;
  box-sizing: border-box;
  background-color: #D9DCE9;
  border-radius: 0.25rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0 c1"
  data-testid="chip"
>
  <p
    class="c2"
  >
    Test
  </p>
</div>
`;
